api/src/middlewares/validate.ts

```diff
@@ -2,18 +2,34 @@ import { Request, Response, NextFunction } from "express";
 import joi from "joi";
 import { ApplicationError } from "../ApplicationError";
 
+const fieldsSchema = joi.object().keys({
+  key: joi.string().required(),
+  type: joi.string().required(),
+  title: joi.string().optional(),
+  answer: joi.any().optional(),
+});
+
 const questionSchema = joi.object().keys({
   index: joi.number().optional(),
   category: joi.string().allow(null).optional(),
   question: joi.string().required(),
-  fields: joi.array().items(joi.any()).required(),
+  fields: joi.array().items(fieldsSchema).required(),
 });
 
 const webhookOutputSchema = joi.object().keys({
   name: joi.string().required(),
   questions: joi.array().items(questionSchema).required(),
-  metadata: joi.object().required(),
   fees: joi.object().optional(),
+  metadata: joi.object({
+    pay: joi
+      .object({
+        payId: joi.string().optional(),
+        state: joi.object().optional(),
+        reference: joi.string().optional(),
+      })
+      .optional(),
+    type: joi.string().trim(),
```

Comment:
Do we want to explicitly outline here that only the form types we have registered in the api are valid form types to be passed through in the metadata i.e "affirmation", "cni", and "exchange"?